panic on concurrent write to JSON-RPC websocket

## panic on concurrent write to JSON-RPC websocket

### Description

If writes are concurrent to the gorilla/websocket writer, the writer panics presumably due to not allowing multiple concurrent writes. 

### Your environment

* OS and version Ubuntu 20
* version of the Polygon SDK ad7b97eeb3a8b66346730b844e0c609992f0efdf
* branch that causes this issue `develop`

### Steps to reproduce

* Tell us how to reproduce this issue <br />
Start in dev mode 

* Where the issue is, if you know <br />

Panic is here in `conn.go` in `gorilla/websocket`:

```go
// Write the buffers to the connection with best-effort detection of
	// concurrent writes. See the concurrency section in the package
	// documentation for more info.

	if c.isWriting {
		panic("concurrent write to websocket connection")
	}
	c.isWriting = true

	err := c.write(w.frameType, c.writeDeadline, c.writeBuf[framePos:w.pos], extra)

	if !c.isWriting {
		panic("concurrent write to websocket connection")
	}
	c.isWriting = false

	if err != nil {
		return w.endMessage(err)
	}

	if final {
		w.endMessage(errWriteClosed)
		return nil
	}
```

* Which commands triggered the issue, if any
`polygon-sdk dev`. Then start a request-heavy client, like a block explorer and point to ws://localhost:8545/ws. You'll eventually hit the error

### Expected behaviour

SDK should not crash

### Actual behaviour

SDK crashes

### Logs

```
Exception has occurred: panic
"concurrent write to websocket connection"
Stack:
	2  0x0000000000eb88f6 in github.com/gorilla/websocket.(*messageWriter).flushFrame
	    at /home/julian/sx-network/polygon-sdk/vendor/github.com/gorilla/websocket/conn.go:610
	3  0x0000000000eb9d65 in github.com/gorilla/websocket.(*Conn).WriteMessage
	    at /home/julian/sx-network/polygon-sdk/vendor/github.com/gorilla/websocket/conn.go:763
	4  0x0000000001010b49 in github.com/0xPolygon/minimal/jsonrpc.(*wsWrapper).WriteMessage
	    at /home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:116
	5  0x000000000101a0f9 in github.com/0xPolygon/minimal/jsonrpc.(*JSONRPC).handleWs.func3
	    at /home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:186
```
More:
```
	    panic: concurrent write to websocket connection

goroutine 111 [running]:
github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc0003a7d78, 0xc000040201, 0xc0000402d0, 0x0, 0x0, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/vendor/github.com/gorilla/websocket/conn.go:610 +0xad6
github.com/gorilla/websocket.(*Conn).WriteMessage(0xc000130840, 0x1, 0xc0000402d0, 0x0, 0x0, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/vendor/github.com/gorilla/websocket/conn.go:763 +0x2e5
github.com/0xPolygon/minimal/jsonrpc.(*wsWrapper).WriteMessage(0xc0006ae980, 0x1, 0xc0000402d0, 0x50, 0x50, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:116 +0x89
github.com/0xPolygon/minimal/jsonrpc.(*JSONRPC).handleWs.func3(0xc000693170, 0xc000542e00, 0x4f, 0x200, 0xc0006ae980, 0x1)
	/home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:186 +0x499
created by github.com/0xPolygon/minimal/jsonrpc.(*JSONRPC).handleWs
	/home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:176 +0x80f
panic: concurrent write to websocket connection

goroutine 166 [running]:
github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc000557d78, 0xc00048c301, 0xc00048c411, 0x0, 0xf, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/vendor/github.com/gorilla/websocket/conn.go:610 +0xad6
github.com/gorilla/websocket.(*Conn).WriteMessage(0xc000130840, 0x1, 0xc00048c411, 0x0, 0xf, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/vendor/github.com/gorilla/websocket/conn.go:763 +0x2e5
github.com/0xPolygon/minimal/jsonrpc.(*wsWrapper).WriteMessage(0xc0006ae980, 0x1, 0xc00048c3c0, 0x51, 0x60, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:116 +0x89
github.com/0xPolygon/minimal/jsonrpc.(*JSONRPC).handleWs.func3(0xc000693170, 0xc000488600, 0x50, 0x200, 0xc0006ae980, 0x1)
	/home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:186 +0x499
created by github.com/0xPolygon/minimal/jsonrpc.(*JSONRPC).handleWs
	/home/julian/sx-network/polygon-sdk/jsonrpc/jsonrpc.go:176 +0x80f

```



### Proposed solution

So there's a note in the code here about concurrency options but it doesn't look it will do much. I've implemented this 

https://github.com/gorilla/websocket/issues/119

here

https://github.com/0xPolygon/polygon-sdk/pull/114

which solves the issue. There doesn't seem to be any performance impact since this is just writing the data to the writer (processing is done way before this).

--- a/.gitignore
+++ b/.gitignore
@@ -14,3 +14,4 @@ main
 
 # MacOS Leftovers
 .DS_Store
+.vscode

--- a/jsonrpc/jsonrpc.go
+++ b/jsonrpc/jsonrpc.go
@@ -5,6 +5,7 @@ import (
 	"io/ioutil"
 	"net"
 	"net/http"
+	"sync"
 
 	"github.com/gorilla/websocket"
 	"github.com/hashicorp/go-hclog"
@@ -101,6 +102,7 @@ var wsUpgrader = websocket.Upgrader{
 type wsWrapper struct {
 	ws     *websocket.Conn // the actual WS connection
 	logger hclog.Logger    // module logger
+	writeLock sync.Mutex // writer lock
 }
 
 // WriteMessage writes out the message to the WS peer
@@ -108,6 +110,8 @@ func (w *wsWrapper) WriteMessage(
 	messageType int,
 	data []byte,
 ) error {
+	w.writeLock.Lock()
+	defer w.writeLock.Unlock()
 	writeErr := w.ws.WriteMessage(messageType, data)
 	if writeErr != nil {
 		w.logger.Error(

